rest-service/manager_rest/rest/resources_v3_1/deployments.py

```diff
@@ -98,6 +99,52 @@ def put(self, deployment_id, **kwargs):
         )
         return deployment, 201
 
+    @authorize('deployment_create')
+    @rest_decorators.marshal_with(models.Deployment)
+    def patch(self, deployment_id):
+        """
+        Update a deployment
+
+        Currently, this function only updates the deployment's labels.
+        """
+        if not request.json:
+            raise IllegalActionError('Update a deployment request must include'
+                                     ' at least one parameter to update')
+        sm = get_storage_manager()
+        deployment = sm.get(models.Deployment, deployment_id)
+
+        _update_labels(sm, deployment)
+
+        return deployment
+
+
+def _update_labels(sm, deployment):
+    """
+    Updating the deployment's labels.
+
+    This function replaces the existing deployment's lables with the new labels
+    that were passed in the request.
+    If a new label already exists, it won't be created again.
+    If an existing label is not in the new labels list, it will be deleted.
+    """
+    new_labels = _get_labels(request.json)
+    if new_labels is None:
+        return
+
+    rm = get_resource_manager()
+    new_labels_set = set(new_labels)
+    existing_labels = deployment.labels
+    existing_labels_tup = set(
+        (label.key, label.value) for label in existing_labels)
+
+    labels_to_create = new_labels_set - existing_labels_tup
+
+    for label in existing_labels:
+        if (label.key, label.value) not in new_labels_set:
+            sm.delete(label)
+
+    rm.create_deployment_labels(deployment, labels_to_create)
```

Comment:
So then how are labels deleted through this?

Reply:
Resolved in slack